core/src/main/java/com/nuodb/migrator/cli/run/CliRunSupport.java

```diff
@@ -152,20 +152,22 @@ protected Group createSourceGroup() {
         OptionFormat optionFormat = new OptionFormat(getOptionFormat());
         optionFormat.setValuesSeparator(null);
 
-        Option driver = newBasicOptionBuilder().withName(SOURCE_DRIVER)
-                .withDescription(getMessage(SOURCE_DRIVER_OPTION_DESCRIPTION)).withRequired(true)
-                .withArgument(newArgumentBuilder().withName(getMessage(SOURCE_DRIVER_ARGUMENT_NAME)).withRequired(true)
-                        .withMinimum(1).withOptionFormat(optionFormat).build())
-                .build();
-        group.withOption(driver);
-
         Option url = newBasicOptionBuilder().withName(SOURCE_URL)
                 .withDescription(getMessage(SOURCE_URL_OPTION_DESCRIPTION)).withRequired(true)
                 .withArgument(newArgumentBuilder().withName(getMessage(SOURCE_URL_ARGUMENT_NAME)).withRequired(true)
                         .withMinimum(1).withOptionFormat(optionFormat).build())
                 .build();
         group.withOption(url);
 
+        Option driver = newBasicOptionBuilder().withName(SOURCE_DRIVER)
+                .withDescription(getMessage(SOURCE_DRIVER_OPTION_DESCRIPTION))
+                .withRequired(false)
+                .withArgument(newArgumentBuilder().withName(getMessage(SOURCE_DRIVER_ARGUMENT_NAME)).withRequired(true)
```

Comment:
I'm confused about the `withRequired(false)... withRequired(true)`. Is it required or not?

Reply:
It looks like that was actually for the `newArgumentBuilder()`.

Reply:
newArgumentBuilder returns an Argument or something like that.  so it's setting argument. I'll have to look at the code more.  Welcome to nuodb-migrator which is a lot of code that is inconsistent throughout.

Reply:
It looks okay.  The option needs to be false, if the option is used then the argument (withArgument) is required.  So the question about withRequired(true) is it is required.   As that is on the Argument not the Option.  The Option is not required.   If you compare before and after only change is the first withRequired().  it was true and is now false.